libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -108,62 +131,40 @@ public static synchronized SyncManager getInstance(UserAccount account, String c
         if (account == null) {
             account = SalesforceSDKManagerWithSmartStore.getInstance().getUserAccountManager().getCurrentUser();
         }
-        if (account == null) {
-            return null;
-        }
-        String uniqueId = account.getUserId();
-        if (UserAccount.INTERNAL_COMMUNITY_ID.equals(communityId)) {
-            communityId = null;
-        }
-        if (!TextUtils.isEmpty(communityId)) {
-            uniqueId = uniqueId + communityId;
-        }
-        SyncManager instance = null;
-        if (INSTANCES == null) {
-            INSTANCES = new HashMap<String, SyncManager>();
-            instance = new SyncManager(account, communityId);
-            INSTANCES.put(uniqueId, instance);
-        } else {
-            instance = INSTANCES.get(uniqueId);
-        }
-        if (instance == null) {
-            instance = new SyncManager(account, communityId);
-            INSTANCES.put(uniqueId, instance);
-        }
-        return instance;
+
+        SmartStore smartStore = SmartSyncSDKManager.getInstance().getSmartStore(account, communityId);
+        SyncManager syncManager = getInstanceForStore(smartStore);
+
+        // Setting up rest client
+        RestClient restClient = SalesforceSDKManager.getInstance().getClientManager().peekRestClient(account);
```

Comment:
Presumably, the developer would manually specify the RestClient for that manager, in that case?